dotCMS/src/main/java/com/dotcms/rest/IntegrityResource.java

```diff
@@ -92,159 +86,149 @@
 
-		cacheEndpointState(endpointId, response.getCookies());
+        final Builder requestBuilder = RestClientBuilder.newClient().target(url).request(mediaType);
 
-		return response;
-	}
+        applyEndpointState(endpoint.getId(), requestBuilder);
 
-    /**
-     * Resolves remote IP address from request.
-     * @param request {@link HttpServletRequest}
-     * @return a String representing the remote IP address (or hostname)
-     */
-    private static String resolveRemoteIp(final HttpServletRequest request) {
-        final String remoteIP = request.getRemoteHost();
-        return !UtilMethods.isSet(remoteIP) ? remoteIP : request.getRemoteAddr();
-    }
+        final Optional<String> requestToken = AuthCredentialPushPublishUtil.INSTANCE.getRequestToken(endpoint);
 
-    /**
-     * Tries to get the local address plus the port in a "host:port" format
-     * @param request http servlet request
-     * @return a string representing the address plus the port
-     */
-    private static String getFullLocalIp(@Context final HttpServletRequest request) {
-        final String localIp = request.getLocalName();
-        final Optional<String> port = HttpRequestDataUtil.getServerPort();
-        return (!UtilMethods.isSet(localIp) ? localIp : request.getLocalName())
-                + ':' + port.orElse(String.valueOf(request.getLocalPort()));
+        if (!requestToken.isPresent()) {
+            Logger.warn(IntegrityResource.class, "No Auth Token set for endpoint:" + endpoint.getId());
+            return response("No Auth Token set for endpoint", true);
+        }
+
+        requestBuilder.header("Authorization", requestToken.get());
+        final Response response = method == HTTPMethod.POST ? requestBuilder.post(entity) : requestBuilder.delete();
+
+        cacheEndpointState(endpoint.getId(), response.getCookies());
+
+        return response;
     }
 
     /**
      * <p>Returns a zip with data from structures and folders for integrity check
      */
     @POST
-    @Path("/generateintegritydata/{params:.*}")
-    @Consumes(MediaType.MULTIPART_FORM_DATA)
+    @Path("/_generateintegritydata")
     @Produces("text/plain")
-    public Response generateIntegrityData(
-            @Context final HttpServletRequest request,
-            @FormDataParam("AUTH_TOKEN") final String auth_token_digest)  {
-        final String localAddress = getFullLocalIp(request);
-        if (!UtilMethods.isSet(auth_token_digest)) {
-            final String message = "Error: Authentication Token was not found.";
-            Logger.error(IntegrityResource.class, String.format("Receiver at %s> :%s", localAddress, message));
-            return Response
-                    .status(HttpStatus.SC_BAD_REQUEST)
-                    .entity(message)
-                    .build();
+    public Response generateIntegrityData(@Context HttpServletRequest request)  {
+
+        final String localAddress = RestEndPointIPUtil.getFullLocalIp(request);
```

Comment:
Codacy found an issue: [Avoid declaring a variable if it is unreferenced before a possible exit point.](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6190239)